compiler/rustc_trait_selection/src/error_reporting/traits/mod.rs

```diff
@@ -127,6 +137,195 @@ pub enum DefIdOrName {
     Name(&'static str),
 }
 
+#[extension(pub trait TypeErrCtxtExt<'a, 'tcx>)]
+impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
```

Comment:
why did you move this out of `fulfillment_errors`?

Reply:
I believe I wanted to separate the public API (`report_fulfillment_errors`) from the impl details (reporting each specific error). Also wanted to make `fulfillment_errors` smaller module, bc it was like 4000 lines long.

Ideally `fulfillment_errors` has all the private impls (reporting specific fulfillment errors) except it's still currently pub because we report a single error (selection errors) manually in coercion rn.